cms/navigation/models.py

```diff
@@ -0,0 +1,107 @@
+from typing import TYPE_CHECKING, ClassVar
+
+from django.db import models
+from django.utils.translation import gettext_lazy as _
+from wagtail.admin.panels import FieldPanel
+from wagtail.blocks import CharBlock, ListBlock, PageChooserBlock, StructBlock
+from wagtail.contrib.settings.models import register_setting
+from wagtail.models import DraftStateMixin, PreviewableMixin, RevisionMixin
+
+from cms.core.blocks.base import LinkBlock
+from cms.core.fields import StreamField
+from cms.core.models import BaseSiteSetting
+
+MAIN_MENU_MAX_NUM_HIGHLIGHTS = 3
+MAIN_MENU_MAX_NUM_COLUMNS = 3
+MAIN_MENU_MAX_NUM_SECTIONS = 3
+MAIN_MENU_MAX_NUM_TOPIC_LINKS = 15
+
+if TYPE_CHECKING:
+    from django.http import HttpRequest
+
+
+class ThemeLinkBlock(LinkBlock):
+    page = PageChooserBlock(required=False, page_type="themes.ThemePage")
+
+    class Meta:
+        label = _("Theme Link")
+
+
+class TopicLinkBlock(LinkBlock):
+    page = PageChooserBlock(required=False, page_type="topics.TopicPage")
+
+    class Meta:
+        label = _("Topic Link")
+
+
+class HighlightsBlock(LinkBlock):
+    description = CharBlock(
+        required=True, max_length=50, help_text=_("For example: 'View our latest and upcoming releases.'")
+    )
+
+    class Meta:
+        icon = "star"
+        label = _("Highlight")
+
+
+class SectionBlock(StructBlock):
+    section_link = ThemeLinkBlock(help_text=_("Main link for this section (Theme pages or external URLs)."))
+    links = ListBlock(
+        TopicLinkBlock(),
+        help_text=_("Sub-links for this section (Topic pages or external URLs)."),
+        max_num=MAIN_MENU_MAX_NUM_TOPIC_LINKS,
+    )
+
+    class Meta:
+        icon = "folder"
+        label = _("Section")
+
+
+class ColumnBlock(StructBlock):
+    sections = ListBlock(SectionBlock(), label="Sections", max_num=MAIN_MENU_MAX_NUM_SECTIONS)
+
+    class Meta:
+        icon = "list-ul"
+        label = _("Column")
+
+
+class MainMenu(DraftStateMixin, RevisionMixin, PreviewableMixin, models.Model):
```

Comment:
I see `MainMenu` now inherits from `RevisionMixin`. Could you please read https://docs.wagtail.org/en/stable/topics/snippets/features.html#saving-revisions-of-snippets and update accordingly?

Reply:
Done. I had rest of the code for it before and then not sure what iteration I lost it.